.github/workflows/image.yml

```diff
@@ -41,27 +49,44 @@ jobs:
 
         docker buildx build \
           --pull \
-          --platform linux/amd64 \
-          --cache-from type=registry,ref=ghcr.io/getsentry/snuba:buildcache \
-          --cache-from type=registry,ref=ghcr.io/getsentry/snuba:latest \
-          --tag ghcr.io/getsentry/snuba:${{ github.sha }} \
+          --platform linux/${{ matrix.arch }} \
+          --cache-from type=registry,ref=ghcr.io/getsentry/snuba:${{ matrix.arch }}-buildcache \
+          --tag ghcr.io/getsentry/snuba:${{ matrix.arch }}-${{ github.sha }} \
           --target application \
           "${args[@]}" \
           .
 
-    - id: get_image_tag
-      run: echo 'image_tag=ghcr.io/getsentry/snuba:${{ github.sha }}' >> "$GITHUB_OUTPUT"
+  assemble:
+    needs: build
+    if: github.event_name != 'pull_request'
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Docker Login
+        run: docker login --username '${{ github.actor }}' --password '${{ secrets.GITHUB_TOKEN }}' ghcr.io
+
+      - name: Assemble Sha Image
+        run: |
+          docker buildx imagetools create -t "ghcr.io/getsentry/snuba:${{ github.sha }}" \
+            "ghcr.io/getsentry/snuba:arm64-${{ github.sha }}" \
+            "ghcr.io/getsentry/snuba:amd64-${{ github.sha }}"
+
+      - name: Assemble Latest Image
+        if: github.ref_name == 'main'
+        run: |
+          docker buildx imagetools create -t "ghcr.io/getsentry/snuba:latest" \
```

Comment:
are you 100% sure that `imagetools create` pushes to the registry? though i don't see a `--push` flag avail

Reply:
Yes it does, this is [taken from Symbolicator](https://github.com/getsentry/symbolicator/blob/master/.github/workflows/image.yml#L171-L182) and that's how it pushes its images.